File: src/cmcandy/java_language_answers/_0104.java
package cmcandy.java_language_answers;

import java.util.LinkedList;

/**
 * Created by CHEN on 2020/6/7.
 */
public class _0104 {
    /**
     * Definition for a binary tree node.
     * public class TreeNode {
     * int val;
     * TreeNode left;
     * TreeNode right;
     * TreeNode(int x) { val = x; }
     * }
     */
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

    class Solution {
        public int maxDepth(TreeNode root) {
            int size = 1;
            TreeNode tmp;
            //利用队列来实现层序遍历
            LinkedList<TreeNode> list = new LinkedList<>( );
            int deep = 0;
            if (root == null) return deep;
            list.add(root);
            while (list.size( ) > 0) {
                //开始遍历
                size = list.size( );
                for (int i = 0; i < size; i++) {
                    tmp = list.removeFirst( );
                    //如果还有下一层，那就入队
                    if (tmp.left != null) {
                        list.add(tmp.left);
                    }
                    if (tmp.right != null) {
                        list.add(tmp.right);
                    }
                }
                deep++;
            }
            return deep;
        }
    }
}

File: src/cmcandy/C_Language_Answers/_0104.cpp
#include <iostream>
#include <math.h>

using namespace ::std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};
class Solution
{
public:
    int maxDepth(TreeNode *root)
    {
        int depth = 0;
        if (root == NULL)
            return 0;
        if (root->left == NULL && root->right == NULL)
            return 1;
        else
            return max(maxDepth(root->left), maxDepth(root->right)) + 1;
    }
};

// void test(TreeNode *root){
//     TreeNode node = TreeNode(2);
//     root = &node;
//     cout<<(root->val)<<endl;
// }
// int main(int argc, char const *argv[])
// {
//     TreeNode node = TreeNode(1);
//     TreeNode* point = &node;
//     cout<<(point->val)<<endl;

//     test(point);
//     cout<<(point->val);
//     return 0;
// }
